### ISSUE TITLE
Unspecified dependencies in package.json

### ISSUE THREAD

**Issue by mkue** [Camberi/firecms#372]
While switching from `yarn` to `npm` due to performance issues with Docker I ran into some dependency problems:

```
Module not found: Error: Can't resolve 'react-fast-compare' in '/home/node/admin/node_modules/@camberi/firecms/dist'
Module not found: Error: Can't resolve '@uiw/react-markdown-preview' in '/home/node/admin/node_modules/@camberi/firecms/dist'
Module not found: Error: Can't resolve '@firebase/auth' in '/home/node/admin/node_modules/@camberi/firecms/dist'
Module not found: Error: Can't resolve '@mui/base' in '/home/node/admin/node_modules/@camberi/firecms/dist'
```

It seems that those packages are not installed but imported/used by firecms. For example [here](https://github.com/Camberi/firecms/blob/next/lib/src/preview/components/Markdown.tsx) or [here](https://github.com/Camberi/firecms/blob/next/lib/src/core/components/SubmitListener.tsx). Does anyone know why this problem appeared while switching from `yarn` to `npm`? And why are the dependencies are not specified in the `package.json` file in the first place?

**Comment by fgatti675** [comment]
Hi, it is normal to have transient dependencies, that are not explicitly listed.
I will check why it is not working with npm

**Comment by mkue** [comment]
Ok, thanks for the quick response! :) What is the reason that you don't list it under `peerDependencies`?

I also had some dependency issues during the `npm` installation. As far as I see it, `@mui/styles` is not compatible with React 18 (https://mui.com/system/styles/basics/) which is why the installation only works with the `--force` flag for me. Don't you get a dependency warning as well?

I didn't spend much time on this and need to investigate a little more. I'll let you know what I find out.

**Comment by fgatti675** [comment]
Yep, cleaning up @mui/styles now since it is not used anymore

**Comment by samwightt** [comment]
Hello, also dealing with this. When using Yarn 2 or PNPM, Vite cannot find the dependencies that aren't explicitly listed under `package.json`. This is because Yarn Berry and pnpm do not allow packages to access dependencies of dependencies. The solution to this is to list the packages imported explicitly in the `package.json`.

**Comment by samwightt** [comment]
@mkue Also getting a dependency warning about `@mui/styles`. Is there a reason why `@mui/styles` is a peer dependency?

**Comment by fgatti675** [comment]
Hi guys, we have made a bunch of dependency fixes in the latest `2.0.0-alpha.28` version.
I have done some testing and it seems to work fine with npm and yarn 3.
Let me know if you are still facing issues,
also keep in mind you should now remove `@mui/styles` from your dependencies

**Comment by mkue** [comment]
Thanks for looking into this so quickly🙏  Everything works now as expected for me.

Feel free to close this.
